sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusSenderClientIntegrationTest.java

```diff
@@ -62,6 +96,47 @@ void nonSessionMessageBatch() {
         for (ServiceBusMessage message : messages) {
             Assertions.assertTrue(batch.tryAdd(message));
         }
+
         sender.send(batch);
+
+        for (int i = 0; i < messages.size(); i++) {
+            messagesPending.incrementAndGet();
+        }
+    }
+
+    void setSenderAndReceiver(MessagingEntityType entityType) {
```

Comment:
I can see that quickly we will have to change this to include other input parameters, for example
queuename / topic names , sessionId for receivers . So instead of taking taking just one parameters `entityType`, it should take a bag of things and  use that  bag to create receivers.

Reply:
There is an overload that takes: `Function<ServiceBusReceiverClientBuilder, ServiceBusReceiverClientBuilder> onCreate`, where you can apply any other custom modifications.